Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -65,12 +59,8 @@ struct Endpoint<H: Handler>: AnyEndpoint {
     /// All `@Parameter` `RequestInjectable`s that are used inside handling `Component`
     var parameters: [AnyEndpointParameter]
     
-    var absolutePath: [_PathComponent] {
-        treeNode.absolutePath
-    }
-    var relationships: [EndpointRelationship] {
-        treeNode.relationships
-    }
+    fileprivate(set) var absolutePath: [_PathComponent] = []
+    fileprivate(set) var relationships: [EndpointRelationship] = []
```

Comment:
Something discussed here: https://github.com/Apodini/Apodini/pull/104#discussion_r553823242

`absolutePath` and `relationships` isn't valid until after the Endpoint got placed into the tree. As illustrated in the linked comment I would love to make that behavior explicit to catch any invalid accesses (which should never happen) immediately.

In the linked PR I added a `fileprivate mutating func onInserted(at:)` to the Endpoint, which handles any necessary initialization once inserted into the tree. Maybe we can coordinate those PRs so something like this is introduced. Would remove the init steps from `addEndpoint` into the `Endpoint` itself, redoing clutter and giving more flexibility I think.